src/VirtoCommerce.ElasticAppSearch.Data/Services/Builders/SearchQueryBuilder.cs

```diff
@@ -68,4 +69,11 @@ protected virtual IFilters GetFilters(ISearchFilter filter, Schema schema)
     {
         return _searchFiltersBuilder.ToFilters(filter, schema);
     }
+
+    protected virtual Dictionary<string, object> GetResultFields(IEnumerable<string> includeFields)
+    {
+        var result = includeFields?.ToDictionary(x => _fieldNameConverter.ToProviderFieldName(x), y => (object)(new { raw = new object() }));

```

Comment:
Move `FieldValue` to the `Search` from `Search/Result` and reuse it for `ResultFields` in `SearchQuery` instead of making anonymous objects